### PR TITLE
Fix fullscreen fill-extrusion querying at low pitch

### PR DESCRIPTION
**Pull request by arindam1993**
The way query geometry was being constructed for fill-extrusions was incorrect for bounds based queries in which the point underneath the camera would fall within the query volume.

This PR fixes it by considering all 4 possible cases
1. The point underneath the camera is not behind the query bounds, in this case ignore it
2. The point is exactly behind, in this case insert it between the last two points
3. The point is behind and to the left, so replace the 4th point
4. The point is behind and to the right, so replace the 3rd point.

I've added some ASCII art code comments to make it clearer.

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality
 - [x] tagged @mapbox/gl-native if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
 - [x] add an entry inside this element for inclusion in the `mapbox-gl-js` changelog: `<changelog>Fill fullscreen fill-extrusion querying at low pitch.</changelog>`

### CODE REVIEW

File: test/integration/query-tests/fill-extrusion/top-down/style.json [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      "debug": true,
+      "width": 200,
+      "height": 400,
+      "queryGeometry": [
+        [0, 0],
+        [200, 400]
+      ]
+    }
+  },
+  "pitch": 0,
```
Comment (tsuz) [inline test/integration/query-tests/fill-extrusion/top-down/style.json:14]: Will this work when pitch is greater than zero?

Comment (arindam1993) [reply test/integration/query-tests/fill-extrusion/top-down/style.json:14]: yep, we already have another test that covers for that, the bug mainly manifested at low pitches.

File: src/style/query_geometry.js [hunk 2]
```diff
@@ -101,21 +101,81 @@ export class QueryGeometry {
...
-        ];
+        const min = this.screenBounds[0];
+        const max = this.screenBounds.length === 1 ? this.screenBounds[0].add(new Point(1, 1)) : this.screenBounds[1];
+        const cameraPolygon = polygonizeBounds(min, max, 0, false);
+
+        // Only need to account for point underneath camera if its behind query volume
+        if (this.cameraPoint.y > max.y) {
+            //case 1: insert point in the middle
+            if (this.cameraPoint.x > min.x && this.cameraPoint.x < max.x) {
+                cameraPolygon.push(this.cameraPoint);
```
Comment (ansis) [inline src/style/query_geometry.js:163]: Would `cameraPolygon.splice(3, 0, this.cameraPoint)` work here?

**Review by ansis — APPROVED** [review]
Looks good to me!

The way `bufferedCameraGeometry` depends on the order of the `polygonizeBounds` return value seemed a bit fragile to me at first but I can't think of a better approach. And the `polygonizeBounds` documentation is very explicit about the order.

One small nit.

File: src/style/query_geometry.js (hunk 2 above)
Comment (arindam1993) [reply src/style/query_geometry.js]: yes derp, I keep forgetting splice can be usedlike that

**Pull request merged**